File: CSharp/Tests/MilkAndCookiesTest.cs
using System;
using Xunit;

namespace CSharp.Tests
{
    public class MilkAndCookiesTest
    {
        [Theory]
        [InlineData(2013, 12, 24, true)]
        [InlineData(2013, 1, 23, false)]
        [InlineData(3000, 12, 24, true)]
        public void TimeForMilkAndCookies_IntMonthDateYearValues_ReturnBoolean(int year, int month, int day, bool expected)
        {
            var actual = MilkAndCookies.TimeForMilkAndCookies(year, month, day);

            Assert.Equal(expected, actual);
        }

        [Fact]
        public void TimeForMilkAndCookies_InvalidDateValues_ReturnArgumentException()
        {
            var year = 2013;
            var month = 0;
            var day = 23;

            Assert.Throws<ArgumentOutOfRangeException>(() => MilkAndCookies.TimeForMilkAndCookies(year, month, day));
        }
    }
}

File: js/tests/milkAndCookies.test.js
const timeForMilkAndCookies = require("../milkAndCookies");

const data = [
  [new Date(2013, 11, 24), true],
  [new Date(3000, 11, 24), true],
  [new Date(2154, 11, 11), false],
  [new Date(2010, 11, 2), false],
  [new Date(1980, 2, 24), false]
];

describe("timeForMilkAndCookies", () => {
  describe("Return true of it is Christmas Eve or not", () => {
    test.each(data)("timeForMilkAndCookies(%o) = %o", (date, expected) => {
      let actual = timeForMilkAndCookies(date);

      expect(actual).toEqual(expected);
    });
  });
});
